### PR TITLE
Allow compilation without OpenSSL and Xerces

### PR DESCRIPTION
**Pull request by ArnaudBienner**

### CODE REVIEW

**Comment by palemieux** [comment]
The following line still refers to XML:

https://github.com/DolbyLaboratories/asdcplib/blob/7e0be2284ad33584df532a60901eb5ac1f88f0b4/src/AS_DCP_MXF.cpp#L312

File: src/AS_02_ACES.cpp [hunk 1]
```diff
@@ -268,6 +274,12 @@ AS_02::ACES::create_4122_type5_id(const byte_t* subject_name, Kumu::fsize_t size
   // Derive the asset ID from the digest. Make it a type-5 UUID
   byte_t buf[Kumu::UUID_Length];
   memcpy(buf, bin_buf, Kumu::UUID_Length);
+#else
+  byte_t buf[Kumu::UUID_Length];
+  Kumu::GenRandomUUID(buf);
```
Comment (IMFTool) [inline src/AS_02_ACES.cpp:279]: This modification makes asdcplib compile without OpenSSL, but the resulting track files would be non-compliant:
Calculation of the SHA-1 hash for Ancillary Resources is required per SMPTE ST 2067-50 and cannot be replaced by generating a random UUID.

Comment (kblinova) [reply src/AS_02_ACES.cpp:279]: Good catch. 
Would it be acceptable to only include the ACES implementation when OpenSSL was defined?

Comment (palemieux) [reply src/AS_02_ACES.cpp:279]: > Would it be acceptable to only include the ACES implementation when OpenSSL was defined?

Definitely not ideal.

What about defining/importing a SHA-1 function if OpenSSL is not used?

Comment (IMFTool) [reply src/AS_02_ACES.cpp:279]: I agree with @palemieux. A SHA-1 implementation would also allow to keep AS-02 TimedText wrapping as a feature.

**Comment by jhursty** [comment]
This PR has been superseded by the new branch sha1_builtin.

**Pull request closed without merging**